hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/S3BucketManagerImpl.java

```diff
@@ -101,34 +101,26 @@ public void createS3Bucket(String userName, String bucketName)
     // anonymous access to bucket where the user name is absent.
     String ozoneVolumeName = formatOzoneVolumeName(userName);
 
-    omMetadataManager.getLock().acquireS3Lock(bucketName);
-    try {
-      String bucket =
-          omMetadataManager.getS3Table().get(bucketName);
-
-      if (bucket != null) {
-        LOG.debug("Bucket already exists. {}", bucketName);
-        throw new OMException(
-            "Unable to create S3 bucket. " + bucketName + " already exists.",
-            OMException.ResultCodes.S3_BUCKET_ALREADY_EXISTS);
-      }
-      String ozoneBucketName = bucketName;
-      createOzoneBucket(ozoneVolumeName, ozoneBucketName);
-      String finalName = String.format("%s/%s", ozoneVolumeName,
-          ozoneBucketName);
+    String bucket = omMetadataManager.getS3Table().get(bucketName);
 
-      omMetadataManager.getS3Table().put(bucketName, finalName);
-    } finally {
-      omMetadataManager.getLock().releaseS3Lock(bucketName);
```

Comment:
Sorry I didn't get why we removed the acquire/release bucket lock. Is the caller now supposed to get the lock?

Reply:
Do you mean s3 bucket lock, as we need to acquire that before creating volume. So, that is acquired in caller in OzoneManager.